### DISCUSSION TITLE
How to set password for new user

### DISCUSSION THREAD

**Discussion by Nadia00485** [opendcim/openDCIM#1559] [Q&A] [answered]
Hi, I'm new to OpenDCIM. After adding a new user, how can they log in if there is no password? How do I set a password for them?

**Comment by alex001x** [comment, marked as answer]
Hi,

To add a user in OpenDCIM, if you use Apache (default installation on Linux) this happens on the terminal
You create an OpenDCIM user "Alex" in OpenDCIM
On Linux terminal - Apache server -
use the command: htpasswd yourDirFolderOpendcim/.htpasswd user
"htpasswd /opt/openDCIM/opendcim/.htpasswd Alex"
then follow the instructions
if  user right error then "sudo htpasswd /opt/openDCIM/opendcim/.htpasswd Alex"
then reload Apache
"sudo systemctl reload apache2"
or restart the Apache service
"sudo systemctl restart apache2"
good job !!!

**Reply by Nadia00485** [reply]
Thank you for providing the solution!!

**Reply by cedgnans** [reply]
@Nadia00485 
Does it work for you?

**Comment by cedgnans** [comment]
@Nadia00485 @alex001x 
I've got the same problem but the this method didn't work for me.

![image](https://github.com/user-attachments/assets/413f736d-818d-4ae4-b4f8-4857c4031114)

**Reply by alex001x** [reply]
Hello,

If the user Cédric is not found, it is because your .htpasswd file is not in the default path folder opendcim (/opt/openDCIM/opendcim/.htpasswd), look at your opendcim installation path, you can also check the apache config file
# sudo vim /etc/apache2/sites-available/opendcim.conf (default path apache)
you can see the line that corresponds: AuthUserFile /opt/openDCIM/.htpasswd to find the path of .htpasswd

be careful when you use htpasswd -c (htpasswd [options] username file) the -c option deletes the password file to create a new one and adds the entry for the user. Be careful, this will delete the existing file without warning if it already exists.
to add an additional password in the existing htpasswd file /path/to/.htpasswd username
You add user cedric in opendcim , next in terminal  (sudo htpasswd /opt/openDCIM/.htpasswd cedric)

Alex.

**Reply by cedgnans** [reply]
You were right @alex001x, but even when I do that, it replaces the old user with the new one. 
Suppose I want to create the user cedric. If I type htpasswd -c /opt/openDCIM/.htpasswd cedric, the old user dcim will disappear and the only user left in the /opt/openDCIM/.htpasswd file will be cedric.

**Reply by alex001x** [reply]
hi cedgnans,

If you type 
htpasswd -c /opt/openDCIM/.htpasswd cedric
The -c option deletes the .htpasswd file and then adds only the user cedric. 

To add the password of a new user and keep the other users write this 

htpasswd /opt/openDCIM/.htpasswd cedric

without the -c option so the file is not deleted
-c = create new file .htpasswd empty

Goodluck, Alex.

**Reply by wilpig** [reply]
@alex001x are you mansplaining how to htpasswd?

https://linux.die.net/man/1/htpasswd

**Reply by alex001x** [reply]
- sorry wilpig, i have to review my basics

Hi cedgnans,

If you type
htpasswd -c /opt/openDCIM/.htpasswd cedric
The -c option rewrites the .htpasswd file with only the user cedric.

To add a new user's password and keep the other users, write this

htpasswd /opt/openDCIM/.htpasswd cedric

Good luck, Alex.

**Reply by cedgnans** [reply]
Thanks
